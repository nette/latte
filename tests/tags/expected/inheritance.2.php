<?php
%A%
final class Template%a% extends Latte\Runtime\Template
{
	public const Blocks = [
		['content' => 'blockContent', 'title' => 'blockTitle', 'sidebar' => 'blockSidebar'],
	];


	public function main(array $ʟ_args): void
	{
%A%
		echo "\n";
		$this->renderBlock('content', get_defined_vars()) /* line %d% */;
		echo "\n";
		$this->renderBlock('sidebar', get_defined_vars()) /* line %d% */;
	}


	public function prepare(): array
	{
		extract($this->params);

		if (!$this->getReferringTemplate() || $this->getReferenceType() === 'extends') {
			foreach (array_intersect_key(['person' => '8'], $this->params) as $ʟ_v => $ʟ_l) {
				trigger_error("Variable \$$ʟ_v overwritten in foreach on line $ʟ_l");
			}
		}
		$this->parentName = 'parent';
		return get_defined_vars();
	}


	/** {block content} on line %d% */
	public function blockContent(array $ʟ_args): void
	{
		extract($this->params);
		extract($ʟ_args);
		unset($ʟ_args);

		echo '	<h1>';
		$this->renderBlock('title', get_defined_vars()) /* line %d% */;
		echo '</h1>

	<ul>
';
		foreach ($people as $person) /* line %d% */ {
			echo '		<li>';
			echo LR\Filters::escapeHtmlText($person) /* line %d% */;
			echo '</li>
';

		}

		echo '	</ul>
';
	}


	/** {block title} on line %d% */
	public function blockTitle(array $ʟ_args): void
	{
		echo 'Homepage ';
	}


	/** {block sidebar} on line %d% */
	public function blockSidebar(array $ʟ_args): void
	{
	}
}
