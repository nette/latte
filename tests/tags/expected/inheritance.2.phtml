<?php
%A%
final class Template%a% extends Latte\Runtime\Template
{
	protected const BLOCKS = [
		['content' => 'blockContent', 'title' => 'blockTitle', 'sidebar' => 'blockSidebar'],
	];


	public function main(): array
	{
%A%
		if ($this->getParentName()) {
			return get_defined_vars();
		}
		$this->renderBlock('content', get_defined_vars()) /* line %d% */;
		echo "\n";
		$this->renderBlock('sidebar', get_defined_vars()) /* line %d% */;
		echo "\n";
%A%
	}


	public function prepare(): void
	{
		extract($this->params);
		$this->parentName = "parent";
		if (!$this->getReferringTemplate() || $this->getReferenceType() === "extends") {
			foreach (array_intersect_key(['person' => '8'], $this->params) as $ʟ_v => $ʟ_l) {
				trigger_error("Variable \$$ʟ_v overwritten in foreach on line $ʟ_l");
			}
		}
	}


	/** {block content} on line %d% */
	public function blockContent(array $ʟ_args): void
	{
		extract($this->params);
		extract($ʟ_args);
		unset($ʟ_args);

		echo '	<h1>';
		$this->renderBlock('title', get_defined_vars()) /* line %d% */;
		echo '</h1>

	<ul>
';
		$iterations = 0;
		foreach ($people as $person) /* line %d% */ {
			echo '		<li>';
			echo LR\Filters::escapeHtmlText($person) /* line %d% */;
			echo '</li>
';
			$iterations++;
		}

		echo '	</ul>
';
	}


	/** {block title} on line %d% */
	public function blockTitle(array $ʟ_args): void
	{
		echo 'Homepage ';
	}


	/** {block sidebar} on line %d% */
	public function blockSidebar(array $ʟ_args): void
	{
	}

}
