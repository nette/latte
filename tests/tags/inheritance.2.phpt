<?php

/**
 * Test: {extends ...} test II.
 */

declare(strict_types=1);

use Tester\Assert;

require __DIR__ . '/../bootstrap.php';


$latte = new Latte\Engine;
$latte->setLoader(new Latte\Loaders\StringLoader([
	'parent' => file_get_contents(__DIR__ . '/templates/parent.latte'),

	'main' => <<<'XX'

		{extends "parent"}

		{block content}
			<h1>{block title}Homepage {/block}</h1>

			<ul>
			{foreach $people as $person}
				<li>{$person}</li>
			{/foreach}
			</ul>
		{/block}

		{block sidebar}{/block}

		XX,
]));

Assert::matchFile(
	__DIR__ . '/expected/inheritance.2.php',
	$latte->compile('main'),
);
Assert::matchFile(
	__DIR__ . '/expected/inheritance.2.html',
	$latte->renderToString('main', ['people' => ['John', 'Mary', 'Paul']]),
);
