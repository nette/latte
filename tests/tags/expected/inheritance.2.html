<!DOCTYPE html>
<head>
	<title>HOMEPAGE </title>
</head>

<body>
	<div id="sidebar">
	</div>

	<div id="content">
	<h1>Homepage </h1>

	<ul>
		<li>John</li>
		<li>Mary</li>
		<li>Paul</li>
	</ul>

	HOMEPAGE


		JOHN
		MARY
		PAUL

	</div>
</body>
</html>
Parent: main/extends
